subsys/net/lib/CMakeLists.txt

```diff
@@ -5,12 +5,17 @@ add_subdirectory_if_kconfig(coap)
 add_subdirectory_if_kconfig(lwm2m)
 add_subdirectory_if_kconfig(socks)
 add_subdirectory_if_kconfig(sntp)
-add_subdirectory_ifdef(CONFIG_DNS_RESOLVER      dns)
 add_subdirectory_ifdef(CONFIG_MQTT_LIB          mqtt)
 add_subdirectory_ifdef(CONFIG_NET_CONFIG_SETTINGS config)
 add_subdirectory_ifdef(CONFIG_NET_SOCKETS       sockets)
 add_subdirectory_ifdef(CONFIG_TLS_CREDENTIALS   tls_credentials)
 
+if (CONFIG_DNS_RESOLVER
+    OR CONFIG_DNS_RESPONDER
```

Comment:
According to the Kconfig, you cannot have CONFIG_DNS_RESPONDER nor CONFIG_LLMNR_RESPONDER without CONFIG_DNS_RESOLVER

What was the config you were trying to apply?